www/src/components/navigation/OnThisPageLinks.tsx

```diff
@@ -0,0 +1,127 @@
+import type { MarkdownHeading } from "astro";
+import { useEffect, useMemo, useState } from "react";
+import { Menu } from "@headlessui/react";
+import clsx from "clsx";
+
+type OnThisPageLinksProps = {
+  headings: MarkdownHeading[];
+  title: string;
+};
+
+export default function OnThisPageLinks({
+  headings,
+  title,
+}: OnThisPageLinksProps) {
+  const [visibleHeading, setVisibleHeading] = useState(title);
+
+  const memoedHeadings = useMemo(() => {
+    return [{ depth: 2, slug: "overview", text: title }, ...headings].filter(
+      ({ depth }) => depth > 1 && depth < 4,
+    );
+  }, [headings, title]);
+
+  useEffect(() => {
+    const articleHeadings = Array.from(
+      document.querySelectorAll("article :is(h1,h2,h3)"),
+    );
+
+    const allObservers = articleHeadings.map((h) => {
+      const observer = new IntersectionObserver(
```

Comment:
Is this observer the same as the TOC one? In that case maybe make it a util and both can use the same?

Or is there conflicts between react and Astro version here?